### PR TITLE
add fieldType

### PR DESCRIPTION
**Pull request by islandryu**
Add FeildType #1

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export type FormProp = {
 export type PropsObject = {
```
Comment (dusanjovanov) [inline src/types.ts:16]: PropsObject is what you pass to the library via the props function, it's not what you get in your component

File: src/types.ts [hunk 2]
```diff
@@ -48,3 +49,11 @@ export type FieldConfig<Context = IndexObject, Props = PropsObject> = {
 };
 
 export type ChangeType = 'value' | 'error' | 'blur';
+
+export type FieldType = {
+  value:any;
+  error:Error;
+  onChange:(value: any) => void; 
+  onBlur: () => void;
+  focusRef:React.RefObject<{focus?: ( () => void) | undefined}>
```
Comment (dusanjovanov) [inline src/types.ts:58]: The object inside the ref coul hold more that just the focus funciton, so there should be [key: string]: any there also

File: src/types.ts [hunk 3]
```diff
@@ -48,3 +49,11 @@ export type FieldConfig<Context = IndexObject, Props = PropsObject> = {
 };
 
 export type ChangeType = 'value' | 'error' | 'blur';
+
+export type FieldType = {
```
Comment (dusanjovanov) [inline src/types.ts:53]: This should be exported in index.tsx

File: src/types.ts (hunk 1 above)
Comment (islandryu) [reply src/types.ts]: That's right.
OK.

**Comment by dusanjovanov** [comment]
Maybe rename FieldType to FieldProp ?

**Comment by islandryu** [comment]
Sure.

**Comment by dusanjovanov** [comment]
Published as @fomrx/formx@0.4.3

**Pull request merged**